libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -235,6 +247,54 @@ export class WorkspaceInstallManager extends BaseInstallManager {
       }
     }
 
+    // For pnpm pacakge manager, we need to handle dependenciesMeta changes in package.json. See more: https://pnpm.io/package_json#dependenciesmeta
+    // If dependenciesMeta settings is different between package.json and pnpm-lock.yaml, then shrinkwrapIsUpToDate return false.
+    if (this.rushConfiguration.packageManager === 'pnpm') {
+      // First build a object for dependenciesMeta settings in package.json
+      // key is the package path, value is the dependenciesMeta info for that package
+      const packagePathToDependenciesMetaInPackageJson: { [key: string]: IDependenciesMetaTable } = {};
+      const commonTempFolder: string = this.rushConfiguration.commonTempFolder;
+
+      for (const rushProject of this.rushConfiguration.projects) {
+        const packageJson: PackageJsonEditor = rushProject.packageJsonEditor;
+        const packageJsonFilePath: string = packageJson.filePath;
+
+        // get the relativePackagePath for a package, to align with the value in pnpm-lock.yaml
+        const relativePackagePath: string = path.relative(
+          commonTempFolder,
+          packageJsonFilePath.replace('/package.json', '')
+        );
```

Comment:
You shouldn't need to call `path.relative` for each project, given that the relative path from the `common/temp` folder to the repo root is the same for all projects. You should be able to calculate the relative path between `commonTempFolder` and the repo root and then concatenate that with the project folder relative path.